pkg/controller/mysqlcluster/mysqlcluster_controller_test.go

```diff
@@ -17,65 +16,264 @@ limitations under the License.
+		// controller k8s client
+		c client.Client
+	)
+
+	BeforeEach(func() {
+		var recFn reconcile.Reconciler
+
+		mgr, err := manager.New(cfg, manager.Options{})
+		Expect(err).NotTo(HaveOccurred())
+		c = mgr.GetClient()
+
+		recFn, requests = SetupTestReconcile(newReconciler(mgr))
+		Expect(add(mgr, recFn)).To(Succeed())
+
+		stop = StartTestManager(mgr)
+	})
+
+	AfterEach(func() {
+		close(stop)
+	})
+
+	Describe("when creating a new mysql cluster", func() {
+		var (
+			expectedRequest reconcile.Request
+			cluster         *api.MysqlCluster
+			secret          *core.Secret
+			components      clusterComponents
+		)
+
+		BeforeEach(func() {
+			name := fmt.Sprintf("cluster-%d", rand.Int31())
+			ns := "default"
+
+			expectedRequest = reconcile.Request{
+				NamespacedName: types.NamespacedName{Name: name, Namespace: ns},
+			}
+
+			secret = &core.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: "the-secret", Namespace: ns},
+				StringData: map[string]string{
+					"ROOT_PASSWORD": "this-is-secret",
+				},
+			}
+
+			cluster = &api.MysqlCluster{
+				ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: ns},
+				Spec: api.MysqlClusterSpec{
+					Replicas:   1,
+					SecretName: secret.Name,
+				},
+			}
+
+			components = []runtime.Object{
+				&apps.StatefulSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-mysql", name),
+						Namespace: cluster.Namespace,
+					},
+				},
+				&core.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-mysql-nodes", name),
+						Namespace: cluster.Namespace,
+					},
+				},
+				&core.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-mysql-master", name),
+						Namespace: cluster.Namespace,
+					},
+				},
+				&core.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-mysql", name),
+						Namespace: cluster.Namespace,
+					},
+				},
+				&core.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-mysql", name),
+						Namespace: cluster.Namespace,
+					},
+				},
+				&policy.PodDisruptionBudget{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-mysql", name),
+						Namespace: cluster.Namespace,
+					},
+				},
+			}
+
+			Expect(c.Create(context.TODO(), secret)).To(Succeed())
+			Expect(c.Create(context.TODO(), cluster)).To(Succeed())
+			Eventually(requests, timeout).Should(Receive(Equal(expectedRequest)))
+
+		})
 
-var expectedRequest = reconcile.Request{NamespacedName: types.NamespacedName{Name: "foo", Namespace: "default"}}
-var depKey = types.NamespacedName{Name: "foo-deployment", Namespace: "default"}
+		AfterEach(func() {
+			// manually delete all created resources because GC isn't enabled in the test controller plane
+			removeAllCreatedResource(c, components)
+			c.Delete(context.TODO(), secret)
+			c.Delete(context.TODO(), cluster)
+		})
 
-const timeout = time.Second * 5
+		It("should reconcile the statefulset", func() {
```

Comment:
No need for this test anymore. It should be tested as a component.

Reply:
The components test does not test if the component is deleted to be recreated. This test should test if the statefulset is deleted to be recreated.